src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/DecompressionHandler.cs

```diff
@@ -211,7 +211,12 @@ public DeflateDecompressedContent(HttpContent originalContent)
             { }
 
             protected override Stream GetDecompressedStream(Stream originalStream) =>
-                new DeflateStream(originalStream, CompressionMode.Decompress);
+                // As described in RFC 2616, the deflate content-coding is actually
+                // the "zlib" format (RFC 1950) in combination with the "deflate"
+                // compression algrithm (RFC 1951).  So while potentially
+                // counterintuitive based on naming, this needs to use ZLibStream
+                // rather than DeflateStream.
+                new ZLibStream(originalStream, CompressionMode.Decompress);
```

Comment:
Do you think there is any reverse compatibility issue here?  Anyone who might desire the old behavior?

Reply:
It's possible. The standard has always been for the deflate-compressed content to be wrapped in a zlib wrapper, but some of the early web servers got it wrong:
https://stackoverflow.com/questions/9170338/why-are-major-web-sites-using-gzip/9186091#9186091
We could make it try both, though that gets complicated.

Reply:
> What is the mode of failure when the server provides deflate stream instead of zlib?

Exception saying the data is in the wrong format / can't be decompressed

> Presumably there are enough servers out there providing only deflate that this bug went unfixed for some time.

From what I've seen, it's more that very few clients actually request deflate because there historically was such confusion about what it meant.  But, yes, it's possible someone somewhere would break.

> I'd hate for us to break clients talking to those servers.

@geoffkizer, opinions on what you'd like to see done here?

Reply:
I think we should make this change. We have time in this release to see if it breaks anyone. If it does, we could add a setting or an app config switch or something. But I think that's pretty unlikely.

Reply:
> From what I've seen, it's more that very few clients actually request deflate

I think it's also that you'd need to request deflate as a higher priority than gzip. I've seen clients request "gzip, deflate" or just "gzip", but almost never "deflate" or "deflate, gzip". So in practice, if you are using compression, you're almost always using gzip.

Reply:
Ok, thanks for the discussion.  I'll stick with just using ZLibStream, and I've marked the PR as a breaking change just to ensure we doc this.

Reply:
Got it, so folks do have a workaround by changing their requested `Transfer-Encoding` to a different spec compliant encoding.  That's better than supporting a non-compliant deflate format.  Thanks for the discussion.